machine-learning/01-openpbta_heatmap.Rmd

```diff
@@ -0,0 +1,383 @@
+
+```{r color_palettes}
+# Palette recommended for those with color vision deficencies - we'll use that 
+# for histologies
+histology_colors <- colorblindr::palette_OkabeIto[2:5]
+sample_annotation_colors <- list(
+  short_histology = c(
+    "LGAT" = histology_colors[1],
+    "Ependymoma" = histology_colors[2],
+    "HGAT" = histology_colors[3],
+    "Medulloblastoma" = histology_colors[4]
+  ),
+  composition = c(
+    "Solid Tissue" = "#A0A0A0",  # light grey
+    "Derived Cell Line" = "#000000"  # black
+  ),
+  tumor_descriptor = c(
+    "Initial CNS Tumor" = "#3333FF",
+    "Progressive" = "#FFFF99",
+    "Recurrence" = "#CCCCFF",
+    "Second Malignancy" = "#000033",
+    "Unavailable" = "#FFFFFF"  # white for missing data
+  )
+)
+```
+
+We need to create a special type of object with `HeatmapAnnotation`.
+
+```{r create_annotation, live = TRUE}
+column_annotation <- HeatmapAnnotation(
+  df = sample_annotation_df,
+  col = sample_annotation_colors
+)
+```
+
+### Values for display
+
+We will z-score the expression values for display.
+This is sometimes called a standardized score.
+
+```{r row_normalize}
+# PLIER::rowNorm z-scores the gene (row) values
+zscores_mat <- PLIER::rowNorm(high_var_mat)
+```
+
+Let's look at the mean and standard deviation of values for a randomly selected gene prior to standardization.
+
+```{r random_row_mean}
+mean(high_var_mat[25, ])
+```
+
+```{r random_row_sd}
+sd(high_var_mat[25, ])
+```
+
+And now those values after standardization.
+
+```{r random_z_mean, live = TRUE}
+mean(zscores_mat[25, ])
+```
+
+This value isn't exact zero but it's very close!
+
+```{r random_z_sd, live = TRUE}
+sd(zscores_mat[25, ])
+```
+
+The standard deviation is now 1.
+That means all the variance values should be equal to one, too.
+
+```{r summary_row_variance}
+summary(matrixStats::rowVars(zscores_mat))
+```
+
+If you're filtering to high variance values, it's important to do that _prior_ to standardizing.
+
+### Heatmap itself!
+
+Okay, now we're ready to make a heatmap complete with annotation bars.
+
+```{r make_heatmap}
+Heatmap(zscores_mat,
+        # The distance metric used for clustering the rows
+        clustering_distance_rows = "pearson",
+        # Linkage method for row clustering
+        clustering_method_rows = "average",
+        # Distance metric for columns
+        clustering_distance_columns = "pearson",
+        # Linkage for columns
+        clustering_method_columns = "average",
+        show_row_names = FALSE,
+        show_column_names = FALSE,
+        # Add annotation bars to the top of the 
+        top_annotation = column_annotation,
+        # This will be used as the label for the color bar
+        # of the cells of the heatmap itself
+        name = "z-score")
+```
+
+```{r save_heatmap_png, live = TRUE}
+png(filename = heatmap_file, 
```

Comment:
We should briefly remind them why we have to make the plot twice, once for screen & once to save.